src/socket/udp.rs

```diff
@@ -258,19 +258,19 @@ impl<'a> Socket<'a> {
         Ok(payload_buf)
     }
 
-    /// Enqueue a packet to be send to a given remote endpoint and pass the buffer
-    /// to the provided closure. The closure then returns the size of the data written
-    /// into the buffer.
+    /// Enqueue a packet to be sent to a given remote endpoint and pass the buffer
+    /// to the provided closure. The closure then returns the either size of the data written
+    /// into the buffer or an error. If an error is returned, no packet is enqueued.
     ///
     /// Also see [send](#method.send).
-    pub fn send_with<F>(
+    pub fn send_with<F, E>(
         &mut self,
-        max_size: usize,
+        at_least: usize,
```

Comment:
I like the old `max_size` name better. The new one sounds like the closure will send "at least X bytes", ie it's a "min size", which is not the case. Could you change it back?